File: go/isogram/isogram.go
package isogram

import "strings"

const testVersion = 1

func IsIsogram(w string) bool {
	charMap := map[rune]int{}
	for _, c := range strings.ToUpper(w) {
		if c == ' ' || c == '-' {
			continue
		}
		if charMap[c] > 0 {
			return false
		} else {
			charMap[c] = 1
		}
	}
	return true
}

File: dart/isogram/lib/isogram.dart
class Isogram {
  bool isIsogram(String s) {
    var m = <int>{};
    for (var rune in s.toLowerCase().runes) {
      if (rune == 32 || rune == 45)
        continue;
      if (m.contains(rune))
        return false;
      else
        m.add(rune);
    }
    return true;
  }
}
